source/fundamentals/builders/filters.txt

```diff
@@ -11,3 +11,534 @@ Filters Builders
+   :copyable: false
+   
+   { "$or" : [
+         "qty" : { "$gt" : 8 }, 
+         "color" : "pink"
+         ]}
+
+The following shows the output of the query using the operator specified above:
+   
+.. code-block:: json
+   :copyable: false
+      
+   { "_id": 2, "color": "purple", "qty": 10, "vendor": ["C", "D"] }
+   { "_id": 5, "color": "yellow", "qty": 11, "vendor": ["A", "B"] }
+   { "_id": 6, "color": "pink", "qty": 5, "vendor": ["C"] }
+
+.. _arrays:
+
+Arrays
+------
+
+The array operators evaluate the array field in a document.
+
+The array operator methods include:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Array Method
+     - Matches
+      
+   * - :java-core-api:`all() <com/mongodb/client/model/Filters.html#all(java.lang.String,java.lang.Iterable)>`
+     - documents if the array field contains every element specified in the query.
+
+   * - :java-core-api:`elemMatch() <com/mongodb/client/model/Filters.html#elemMatch(java.lang.String,org.bson.conversions.Bson)>`
+     - documents if an element in the array field matches all the specified conditions.
+      
+   * - :java-core-api:`size() <com/mongodb/client/model/Filters.html#size(java.lang.String,int)>`
+     - documents if the array field is a specified number of elements.
+
+The following example matches documents with a ``vendors`` array
+containing both "A" and "D" in the ``paint_purchases`` collection: 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Filters.java
+   :language: java
+   :dedent:
+   :start-after: begin allComparison
+   :end-before: end allComparison
+   
+The JSON output renders as the following:
+
+.. code-block:: json
+   :copyable: false
+   
+   { "vendor" : { "$all" : { ["A", "D"] }}}
+
+The following shows the output of the query using the operator specified above:
+   
+.. code-block:: json
+   :copyable: false
+      
+   { "_id": 8, "color": "orange", "qty": 7, "vendor": ["A", "D"] }
+
+.. _elements:
+
+Elements
+--------
+
+The elements operators evaluate the nature of a specified field.
+
+The elements operator methods include:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Elements Method
+     - Matches
+      
+   * - :java-core-api:`exists() <com/mongodb/client/model/Filters.html#exists(java.lang.String)>`
+     - documents that have the specified field.
+
+   * - :java-core-api:`type() <com/mongodb/client/model/Filters.html#type(java.lang.String,java.lang.String)>`
+     - documents if a field is of the specified type.
+
+The following example matches documents that have a ``qty`` field and
+its value does not equal "5" or "8" in the ``paint_purchases`` collection:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Filters.java
+   :language: java
+   :dedent:
+   :start-after: begin existsComparison
+   :end-before: end existsComparison
+
+The JSON output renders as the following:
+
+.. code-block:: json
+   :copyable: false
+   
+   { "qty" : [ { 
```

Comment:
In the 4.0 driver it currently renders as:

    {"$and": [{"qty": {"$exists": true}}, {"qty": {"$nin": [5, 8]}}]}

You can see with this code:

        var filter = and(exists("qty"), nin("qty", 5, 8));
        var doc = filter.toBsonDocument(BsonDocument.class, MongoClientSettings.getDefaultCodecRegistry());
        System.out.println(doc.toJson());

But it's subject to change given bugs like https://jira.mongodb.org/browse/JAVA-4006.  So we may need to mention that.